### PR TITLE
Upgraded some libraries for 6.3

### PR DESCRIPTION
**Pull request by MarekB01**
Upgraded Spring boot to 2.7.3 and Hibernate to 5.6.11

Link to QA https://github.com/BroadleafCommerce/QA/issues/4786

### CODE REVIEW

**Review by oleksiimiroshnyk — APPROVED** [review]

File: common/pom.xml [hunk 1]
```diff
@@ -452,5 +452,9 @@
             <groupId>com.googlecode.htmlcompressor</groupId>
             <artifactId>htmlcompressor</artifactId>
         </dependency>
+        <dependency>
```
Comment (oleksiimiroshnyk) [inline common/pom.xml]: Why did you add new dependency?

File: admin/broadleaf-open-admin-platform/src/main/java/org/broadleafcommerce/openadmin/server/dao/provider/metadata/MapFieldsFieldMetadataProvider.java [hunk 2]
```diff
@@ -171,7 +170,6 @@ public MetadataProviderResponse addMetadataFromFieldType(AddMetadataFromFieldTyp
                         ((BasicFieldMetadata) entry.getValue()).getExplicitFieldType(),
                         myType.getReturnedClass(), addMetadataFromFieldTypeRequest.getDynamicEntityDao()), metadata);
 
-                TypeConfigurationRegistry.INSTANCE.deregisterTypeConfiguration(typeConfiguration);
```
Comment (oleksiimiroshnyk) [inline admin/broadleaf-open-admin-platform/src/main/java/org/broadleafcommerce/openadmin/server/dao/provider/metadata/MapFieldsFieldMetadataProvider.java:174]: is this class missing ? or why did you delete it ?

Comment (MarekB01) [reply admin/broadleaf-open-admin-platform/src/main/java/org/broadleafcommerce/openadmin/server/dao/provider/metadata/MapFieldsFieldMetadataProvider.java:174]: It is missing, because Hibernate was upgraded to 5.6.11

File: core/broadleaf-profile/src/main/java/org/broadleafcommerce/profile/core/domain/Country.java [hunk 3]
```diff
@@ -33,6 +33,7 @@
  *
  * @author Elbert Bautista (elbertbautista)
  */
+@Deprecated
```
Comment (riteshadhikari17) [inline core/broadleaf-profile/src/main/java/org/broadleafcommerce/profile/core/domain/Country.java]: @oleksiimiroshnyk Shouldn't we remove this instead of deprecating? I thought [this ticket](https://github.com/BroadleafCommerce/QA/issues/4509) was for removing. Its 6.3 so its better remove it and in Migration notes mention to use `org.broadleafcommerce.common.i18n.domain.ISOCountryImpl`

Comment (riteshadhikari17) [reply core/broadleaf-profile/src/main/java/org/broadleafcommerce/profile/core/domain/Country.java]: @MarekB01 I wasn't talking about removing deprecation tag. I was asking if we should remove these `Country` and `CountryImpl` classes. 
BTW in future, remember to mention the class that should be used along with deprecation tag.

File: core/broadleaf-profile/src/main/resources/bl-profile-applicationContext-entity.xml [hunk 4]
```diff
@@ -23,9 +23,9 @@
            
     <!-- Entity mappings - override for extensibility -->
     <bean id="org.broadleafcommerce.profile.core.domain.Address" class="org.broadleafcommerce.profile.core.domain.AddressImpl" scope="prototype">
-        <property name="country">
-            <bean class="org.broadleafcommerce.profile.core.domain.CountryImpl">
-                <property name="abbreviation" value="US"/>
+        <property name="isoCountryAlpha2">
```
Comment (oleksiimiroshnyk) [inline core/broadleaf-profile/src/main/resources/bl-profile-applicationContext-entity.xml:26]: I think this can be reverter ? as we will not delete CountryImpl

File: core/broadleaf-profile/src/main/java/org/broadleafcommerce/profile/core/domain/Country.java (hunk 3 above)
Comment (oleksiimiroshnyk) [reply core/broadleaf-profile/src/main/java/org/broadleafcommerce/profile/core/domain/Country.java]: @riteshadhikari17 after investigation we decided to not remove CountryImpl, looks like its main purpose is to be "supported country dictionary" while IsoCountry is a global dictionary.
So for now lets leave as it is. 

If we want we can create a separate task to get rid of CountryImpl completely and move it is goal to, let's say, properties or somewhere else.

File: core/broadleaf-profile/src/main/resources/bl-profile-applicationContext-entity.xml (hunk 4 above)
Comment (MarekB01) [reply core/broadleaf-profile/src/main/resources/bl-profile-applicationContext-entity.xml:26]: It cannot see the property with name "country" I think the field Country was removed with State in AddressImpl. I think in this case it would be better to leave the changed code
